File: leetcode/easy/414. Third Maximum Number/solution.cc
#include <vector>

using namespace std;

const long MINIMUM = -9876543210;
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        long f = MINIMUM, s = MINIMUM, t = MINIMUM;

        for (int n: nums) {
            if (n == f || n == s || n == t) continue;
            if (n > f) {
                t = s;
                s = f;
                f = n;
            }
            else if (n > s) {
                t = s;
                s = n;
            }
            else if (n > t) {
                t = n;
            }
        }
        return t == MINIMUM ? f : t;
    }
};

File: leetcode/easy/414. Third Maximum Number/solution.js
// https://leetcode.com/problems/third-maximum-number/
/**
 * @param {number[]} nums
 * @return {number}
 */
function thirdMax(nums) {
    const set = new Set(nums.sort((a, b) => b - a));
    const arr = Array.from(set);
    return arr.length < 3 ? arr[0] : arr[2];
};

thirdMax([1, 2])
// thirdMax([1, 2, 7, 34, 78, 9, 8, 8, 23, 55])
// thirdMax([2, 2, 3, 1])